crates/decoder/examples/cli.rs

```diff
@@ -218,18 +232,19 @@ fn read_flat_file(path: &str, compression: Compression) -> Result<Vec<Block>, De
 /// Dbin file type extension
 const EXTENSION: &str = "dbin";
 
-/// Checks if the file extension is `.dbin`.
+/// Checks if the file extension is `.dbin`, ignoring a .zst extension.
 fn dir_entry_extension_is_dbin(entry: &DirEntry) -> bool {
-    entry
-        .path()
-        .extension()
-        .map_or(false, |ext| ext == EXTENSION)
+    let mut path = entry.path();
+    if path.extension().and_then(|e| e.to_str()) == Some("zst") {
+        path = path.with_extension("");
+    }
+    path.extension().map_or(false, |ext| ext == EXTENSION)
```

Comment:
Rather than complicate things by mutating that variable we can make things a tad more functional this way by creating a new `effective_path` variable that encapsulates stripping the `"zst"` extension if necessary:

```suggestion
    let path = entry.path();
    let effective_path = if path.extension().and_then(|e| e.to_str()) == Some("zst") {
        path.with_extension("")
    } else {
        path
    };

    effective_path.extension().map_or(false, |ext| ext == EXTENSION)
```